src/http/sr-modules-keys/sr-modules-keys.e2e-spec.ts

```diff
@@ -0,0 +1,484 @@
+      });
+    });
+  });
+
+  describe('The /modules/keys request', () => {
+    describe('api ready to work', () => {
+      beforeAll(async () => {
+        // lets save meta
+        await elMetaStorageService.update(elMeta);
+        // lets save keys
+        await keysStorageService.save(keys);
+        // lets save modules
+        await moduleStorageService.upsert(dvtModule, 1);
+        await moduleStorageService.upsert(curatedModule, 1);
+      });
+
+      afterAll(async () => {
+        await cleanDB();
+      });
+
+      it('Should return all keys for request without filters', async () => {
+        // Get all keys and operators without filters
+        const resp = await request(app.getHttpServer()).get(`/v1/modules/keys`);
+
+        expect(resp.status).toEqual(200);
+        expect(resp.body.data).toEqual(expect.arrayContaining(keysByModules));
+        expect(resp.body.meta).toEqual({
+          elBlockSnapshot: {
+            blockNumber: elMeta.number,
+            blockHash: elMeta.hash,
+            timestamp: elMeta.timestamp,
+          },
+        });
+      });
+
+      it('Should return 400 error if operatorIndex is not a number', async () => {
+        const resp = await request(app.getHttpServer())
+          .get(`/v1/modules/keys`)
+          .query({ used: false, operatorIndex: 'one' });
+        expect(resp.status).toEqual(400);
+        expect(resp.body).toEqual({
+          error: 'Bad Request',
+          message: ['operatorIndex must not be less than 0', 'operatorIndex must be an integer number'],
+          statusCode: 400,
+        });
+      });
+
+      it('Should return 400 error if used is not a boolean value', async () => {
+        const resp = await request(app.getHttpServer()).get(`/v1/modules/keys`).query({ used: 0, operatorIndex: 2 });
+        expect(resp.status).toEqual(400);
+        expect(resp.body).toEqual({ error: 'Bad Request', message: ['used must be a boolean value'], statusCode: 400 });
+      });
+
+      it('Should return used keys and operator one', async () => {
+        const resp = await request(app.getHttpServer()).get(`/v1/modules/keys`).query({ used: true, operatorIndex: 1 });
+
+        const expectedKeysDvt = dvtModuleKeys.filter((key) => key.used && key.operatorIndex == 1);
+        const expectedKeysCurated = curatedModuleKeys.filter((key) => key.used && key.operatorIndex == 1);
+
+        expect(resp.status).toEqual(200);
+        expect(resp.body.data).toEqual(
+          expect.arrayContaining([
+            { keys: expectedKeysDvt, module: dvtModuleResp },
+            { keys: expectedKeysCurated, module: curatedModuleResp },
+          ]),
+        );
+        expect(resp.body.meta).toEqual({
+          elBlockSnapshot: {
+            blockNumber: elMeta.number,
+            blockHash: elMeta.hash,
+            timestamp: elMeta.timestamp,
+          },
+        });
+      });
+
+      it('Should return unused keys and operator one', async () => {
```

Comment:
yes, i think to have operatorIndex here as filter is odd scenario. should we leave only 'used' filter for this endpoint?

Reply:
Yeah, that makes sense to me. But if we break backwards compatibility, I'd leave it as is. Was this filter in the previous version?

Reply:
yes, we had this filter
but i don't think someone used it

Reply:
> The test name says it will return keys and operator, but it returns only keys and module info

Still wrong test description. Apparently copied from `sr-modules-operators-keys.e2e-spec.ts`

Reply:
> yes, we had this filter
> but i don't think someone used it

we can't be sure, I would proceed from the rule that if we break backward compatibility, we go through the whole process correctly: new version of endpoint, notification of deprecation, deprecation through several versions